File: data/EXP01/python/2502/2502.output_2.py
# @lc app=leetcode id=2502 lang=python3
class Allocator:

    def __init__(self, n: int):
        self.memory = [None] * n

    def allocate(self, size: int, mID: int) -> int:
        for i in range(len(self.memory)-size+1):
            if all(self.memory[i+j] is None for j in range(size)):
                for j in range(size):
                    self.memory[i+j] = mID
                return i
        return -1

    def free(self, mID: int) -> int:
        count = 0
        for i in range(len(self.memory)):
            if self.memory[i] == mID:
                self.memory[i] = None
                count += 1
        return count


loc = Allocator(10)
print(loc.allocate(1, 1))  # Output: 0
print(loc.allocate(1, 2))  # Output: 1
print(loc.allocate(1, 3))  # Output: 2
print(loc.free(2))  # Output: 1
print(loc.allocate(3, 4))  # Output: 3
print(loc.allocate(1, 1))  # Output: 1
print(loc.allocate(1, 1))  # Output: 6
print(loc.free(1))  # Output: 3
print(loc.allocate(10, 2))  # Output: -1
print(loc.free(7))  # Output: 0

File: data/EXP01/go/2502/2502.output_2.go
/*
* @lc app=leetcode id=2502 lang=golang
*
* [2502] Design Memory Allocator
*/

type Allocator struct {
    memory []int
    used   map[int]struct{}
}

func Constructor(n int) Allocator {
    return Allocator{
        memory: make([]int, n),
        used:   make(map[int]struct{}),
    }
}

func (this *Allocator) Allocate(size int, mID int) int {
    for i := 0; i <= len(this.memory)-size; i++ {
        found := true
        for j := i; j < i+size; j++ {
            if this.memory[j] != 0 {
                found = false
                break
            }
        }
        if found {
            for j := i; j < i+size; j++ {
                this.memory[j] = mID
            }
            return i
        }
    }
    return -1
}

func (this *Allocator) Free(mID int) int {
    count := 0
    for i := 0; i < len(this.memory); i++ {
        if this.memory[i] == mID {
            this.memory[i] = 0
            count += 1
        }
    }
    delete(this.used, mID)
    return count
}

/**
 * Your Allocator object will be instantiated and called as such:
 * obj := Constructor(n);
 * param_1 := obj.Allocate(size,mID);
 * param_2 := obj.Free(mID);
 */
